hw/ip/pwrmgr/data/pwrmgr_testplan.hjson

```diff
@@ -11,22 +11,225 @@
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: aborted_lowpower
+      desc: '''
+            Test aborted low power transitions.
+
+            Low power transitions can be aborted in two cases:
+            - The processor gets an interrupt soon after a low power entry is
+              triggered.
+            - OTP, LC, or FLASH are not idle.
+            This test aborts low power transitions, and disables any wakeups,
+            so the test would timeout if low power was entered.
+
+            **Stimulus**:
+            - Bring pwrmgr to low power.
+            - Either disable `pwr_cpu.core_sleeping` or keep some of `lc_idle`,
+              `otp_idle`, or `flash_idle` inputs off.
+            - Disable all wakeup enables.
+            - Randomly set `wakeup_info_capture_dis` CSR.
+
+            **Checks**:
+            - The `ctrl_cfg_regwen` CSR reads as 1 on the first attempt.
+            - Checks the output `pwr_rst_req.reset_cause` doesn't change for
+              a bounded amount of time.
+            - Check that the `wakeup_info` CSR flags either `fall_through` or
+              `abort` events when capture is enabled.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: reset
+      desc: '''
+            Test random reset and reset_en.
+
+            The different reset inputs can be disabled via bits in the
+            `reset_en` CSR. Resets can be triggered either in active or
+            low power state.
+
+            **Stimulus**:
+            - Sets `reset_en` randomly.
+            - Randomly choose whether to put the unit in low power mode.
+            - Set `rstreqs_i` inputs randomly.
+
+            **Checks**:
+            - Waits for fast fsm to become active reading `ctrl_cfg_regwen`
+              until it becomes 1.
+            - Checks the `reset_status` CSRs.
+            - Checks `ip_clk_en` output has a low transition.
+            - Checks the output `pwr_rst_req.reset_cause` matches a reset
+              cause.
+            - Checks the output `pwr_rst_req.rstreqs` matches the enabled
+              resets.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: escalation_reset
+      desc: '''
+            Test escalation reset.
+
+            An escalation reset cannot be disabled.
+
+            **Stimulus**:
+            - Set `esc_rst_tx_i` inputs randomly.
+
+            **Checks**:
+            - Waits for fast fsm to become active reading `ctrl_cfg_regwen`
+              until it becomes 1.
+            - Checks the `reset_status` CSRs.
+            - Checks the output `pwr_rst_req.reset_cause` matches a reset
+              cause.
+            - Checks the output `pwr_rst_req.rstreqs` is not set.
+            - Check the output `esc_rst_rx_o` indicates an alert escalation.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: reset_wakeup_race
```

Comment:
can we also test low_power_reset race? ie the wake is coming really close to the window where we decide to enter low power. 

 this is probably part of your plan already, but we want to make sure also the pwrmgr doesn't end up in some weird state.

Reply:
Good idea. I added a lowpower_wakeup_race testpoint.